src/read.c

```diff
@@ -1527,6 +1579,62 @@ void ReadFuncExpr1 (
 
 /****************************************************************************
 **
+*F  ReadFuncExprLong(<follow>) . . . . . read a multi-arg function expression
+**
+**  'ReadFuncExprLong' reads  an abbreviated  function literal expression.  In
+**  case of an error it skips all symbols up to one contained in <follow>.
+**
+**      <Function>      := '{' <ArgList> '}' '->' <Expr>
+*/
+void ReadFuncExprLong (
+    TypSymbolSet        follow )
+{
+    volatile Int        narg;           /* number of arguments             */
+    volatile Obj        nams;           /* list of local variables names   */
+
+    Match( S_LBRACE, "{", follow );
+
+    struct ArgList args = ReadFuncArgList(follow, 0, S_RBRACE, ")");
+    narg = args.narg;
+    nams = args.nams;
+
+    /* 'function( a,b, p... )' takes a variable number of arguments           */
+    /* Also, we special case function(arg)                                   */
+    if (args.isvarg) {
+      narg = -narg;
+    }
+
+    ReadFuncExprBody(follow, nams, narg);
+}
+
+/****************************************************************************
+**
+*F  ReadFuncExpr1(<follow>) . . . . . . . . . . .  read a function expression
+**
+**  'ReadFuncExpr1' reads  an abbreviated  function literal   expression.  In
+**  case of an error it skips all symbols up to one contained in <follow>.
+**
+**      <Function>      := <Var> '->' <Expr>
+*/
+void ReadFuncExpr1 (
+    TypSymbolSet        follow )
+{
+    volatile Obj        nams;           /* list of local variables names   */
+    volatile Obj        name;           /* one local variable name         */
```

Comment:
I don't see how that would be related either... We could just try it by making a PR which removes all volatile, and see if anything breaks :) See PR #1084